scenarios/test_codegen/test/HyperSyncWorker_test.res

```diff
@@ -0,0 +1,151 @@
+open RescriptMocha
+
+module GetNextPage = {
+  type queryLogsPageCall = {
+    serverUrl: string,
+    fromBlock: int,
+    toBlock: int,
+    logSelections: array<LogSelection.t>,
+    fieldSelection: HyperSyncClient.QueryTypes.fieldSelection,
+    nonOptionalBlockFieldNames: array<string>,
+    nonOptionalTransactionFieldNames: array<string>,
+  }
+
+  type mock = {queryLogsPageCalls: array<queryLogsPageCall>}
+
+  let mock = (~blockSchema=S.object(_ => ()), ~transactionSchema=S.object(_ => ())) => {
```

Comment:
This mock is not the most flexible, but can be extended to conveniently test the logSelections logic, as well as polling logic